youtube_dl/extractor/mixcloud.py

```diff
@@ -122,3 +120,31 @@ def _real_extract(self, url):
             'view_count': view_count,
             'like_count': like_count,
         }
+
+class ServerAttemptSequence:
```

Comment:
```python
def server_sequence(start, max_value):
     low = high = start
     yield start
     while low > 1 or high < max_value:
         low -= 1
         high += 1
         if high <= max_value:
             yield high
         if low >= 1:
             yield low
```
In the second form, you would need to use `itertools.izip_longest`.

(By the way, you should probably add a docstring explaining what it does so that someone reading it understand what it's supposed to do).